components/p3a/brave_p3a_service.cc

```diff
@@ -99,21 +136,90 @@ BraveP3AService::~BraveP3AService() = default;
 void BraveP3AService::RegisterPrefs(PrefRegistrySimple* registry,
                                     bool first_run) {
   BraveP3ALogStore::RegisterPrefs(registry);
-  registry->RegisterTimePref(kLastRotationTimeStampPref, {});
+  // Using "year ago" as default value to fix macOS test crashes
```

Comment:
What was the reason it was crashing?

Reply:
This was a little while ago, but if I recall correctly, there was a mysterious crash with a stack trace that used the value from this pref. I cannot remember the exact reason for the crash. I don't use macOS for my dev env, so I was trying different methods to make macOS CI happy.

Reply:
If you can get the call stack for me lets work out why

Reply:
I'm pretty certain the crash was from here https://source.chromium.org/chromium/chromium/src/+/main:base/time/time.cc;drc=873dba838a8c8a7a08863097eb71d17b85451da6;l=244

`NextDay` was calling `LocalMidnight`, and it crashed there. Maybe because the default `base::Time` value on macOS is not a valid date?

Reply:
cc @bridiver @iefremov as not sure I have seem this fail, maybe you know why